module/VuFind/src/VuFind/ILS/Driver/AbstractAPI.php

```diff
@@ -98,6 +98,26 @@ public function makeRequest($method = "GET", $path = "/", $params = [],
             } else {
                 $client->setParameterPost($params);
             }
+            if ($this->logger) {
+                // remove passwords
```

Comment:
@crhallberg, these are somewhat FOLIO-specific details; I wonder if it would be better to make a protected debugRequest method that gets called here so that ILS-specific details can be appropriately overridden in subclasses.

Reply:
That's an excellent idea

Reply:
Looks great! My last suggestion would be to make the default implementation a little safer so nothing slips through the cracks. I'd suggest that we don't output POST data or headers at all by default (or maybe just give a count, or show only keys but not values). That will prevent the case where somebody subclasses this, doesn't realize the debug method exists, and accidentally leaks something sensitive.